### DISCUSSION TITLE
Pushing new changes before PR approved

### DISCUSSION THREAD

**Discussion by ckunchur** [CS342/.github#15] [Technical Questions] [answered]
### In what area do you have a technical challenge?

Xcode, Simulator & Previews

### Description

I added the REUSE header to some of my files that were missing it, but I can't push these changes to my branch because my previous PR has not been approved. 

### Reproduction

N/A

### Expected behavior

N/A

### Code of Conduct

- [X] I agree to follow this course's Code of Conduct and Contributing Guidelines

**Comment by vishnuravi** [comment, marked as answer]
Hi Caitlin,

You can continue to push to your feature branch even though the PR hasn't been approved. The PR will be automatically updated!

Vishnu

**Reply by ckunchur** [reply]
Great thanks! Also, we don't need to do anything other than adding the header to meet the licensing requirement right? Paul and Oliver had license files for their profile picture and bio, but I have a lot of resources and was hoping to not have to create a new file for each one.

**Reply by vishnuravi** [reply]
Unfortunately the REUSE spec requires each file to have a license so you would have to create a license file for each image the same way as the jpegs in the example.

This is important because images are often licensed and/or copyrighted differently.

**Reply by ckunchur** [reply]
Aw okay thank you! I just need to append the exact filename before .license right? Would this require me to look up the licensing info for each of my ~15 images individually? I can't just copy the MIT license header right?

**Reply by vishnuravi** [reply]
> Aw okay thank you! I just need to append the exact filename before .license right? Would this require me to look up the licensing info for each of my ~15 images individually? I can't just copy the MIT license header right? 

In your team's project, which will be distributed on the App Store, it's very important that images are licensed and attributed properly. For the purposes of this homework assignment we won't be sticklers about the REUSE files, but you shouldn't include any images you don't have the right to use/distribute.

**Reply by ckunchur** [reply]
Got it! I will be diligent about proper licensing and use for the actual project. For the purposes of the assignment though, if I'm not able to find the license info for all the images/have the MIT license header is that okay?

**Reply by vishnuravi** [reply]
The script we're using for grading the assignment only checks to see if you have a REUSE header for each file, it doesn't have any way of verifying whether the information in it is accurate.

**Reply by PSchmiedmayer** [reply]
@vishnuravi Great summary. As you noted, it is essential for open-source projects to be sure that all content is provided under the enclosing license. Please only use images that you are free to use in the application as they are, e.g., in the public domain or you have copyright over (e.g., an image of you that you took). You can then define that copyright in the `.license` file associated with the image so this is transparent to everyone.

@ckunchur If this thread solved your question, feel free to tag @vishnuravi's comment as an answer, so we keep track of which discussions still need to follow up work and other students can easily spot the solution 👍
